### PR TITLE
Developer page improvements and day summary webhook

### PR DESCRIPTION
**Pull request by mdmohsin7**
- [x] Added day summary webhook
- [x] Added webhook status toggles
- [x] Changed developer page UI
- [x] Updated the logic to trigger only if the toggles are enabled
- [x] Added some more entries to redis and some funcs

https://github.com/user-attachments/assets/3306a629-987f-4930-9b2e-c7257c87600e

## Release Notes

- **New Features**
  - Introduced functions to manage webhooks: enable, disable, and check status.
  - Added new settings for user preferences, including toggles for various webhook functionalities.
  - Enhanced the Developer Settings page with a more user-friendly layout and loading indicators.
  - Added a new `ToggleSectionWidget` for better management of toggleable sections in the settings.

- **Bug Fixes**
  - Improved error handling and validation for webhook URLs.

- **Documentation**
  - Updated documentation to reflect new webhook management capabilities.

- **Style**
  - Streamlined UI components for better organization and visual appeal.

### CODE REVIEW

File: backend/database/redis_db.py [hunk 1]
```diff
@@ -226,6 +226,26 @@ def get_in_progress_memory_id(uid: str) -> str:
 
 def set_user_webhook_db(uid: str, wtype: str, url: str):
     r.set(f'users:{uid}:developer:webhook:{wtype}', url)
+    if url == '' or url == ',':
```
Comment (beastoin) [inline backend/database/redis_db.py]: don't put the ambiguous logic here, move to util / router if needed.

File: backend/database/redis_db.py [hunk 2]
```diff
@@ -226,6 +226,26 @@ def get_in_progress_memory_id(uid: str) -> str:
...
+
+def enable_user_webhook_db(uid: str, wtype: str):
+    r.set(f'users:{uid}:developer:webhook_status:{wtype}', str(True).lower())
+
+def user_webhook_status_db(uid: str, wtype: str):
+    status = r.get(f'users:{uid}:developer:webhook_status:{wtype}')
+    if status is None:
+        res = get_user_webhook_db(uid, wtype)
+        if res != '' and res != ',':
+            enable_user_webhook_db(uid, wtype)
```
Comment (beastoin) [inline backend/database/redis_db.py]: similarly, don't put to many ambiguous logic to the db layer, move to utils pls

**Review by beastoin — COMMENTED** [review]
overall, too many reads to the database. that's not good. please create new ticket to improve it @mdmohsin7

**Comment by mdmohsin7** [comment]
> overall, too many reads to the database. that's not good. please create new ticket to improve it @mdmohsin7

Alright will do

**Comment by beastoin** [comment]
resolve conflicts pls sir ~

**Comment by mdmohsin7** [comment]
> resolve conflicts pls sir ~

Done sir

**Comment by beastoin** [comment]
lgmt @mdmohsin7 🚀

**Pull request merged**